### DISCUSSION TITLE
Building with setuptools?

### DISCUSSION THREAD

**Discussion by barower** [semuconsulting/pynmeagps#44] [General]
I'm trying to write a yocto recipe for `pynmeagps`, but unfortunately none of the python building classes seem to work. `python_setuptools_build_meta` seems most appropriate based on your `pyproject.toml`, but I always end up having following result files:

```
$ tree
.
└── usr
    └── lib
        └── python3.10
            └── site-packages
                └── UNKNOWN-0.0.0.dist-info
                    ├── LICENSE
                    ├── METADATA
                    ├── RECORD
                    ├── top_level.txt
                    └── WHEEL
```

In yocto kirkstone, `setuptools`'s version is 59.5.0. Downgrading `pynmeaps` to 1.0.23, which declares to be using older version of `setuptools` result in same story

What commands should be executed to build the package? This way maybe I could be able to tweak the recipe

With kind regards

**Comment by semuadmin** [comment]
Hi @barower ,

I'm not too familiar with yocto but assuming it deploys a standard Python stack, I see no reason why a conventional build command would not work. Have a look at the build task in the /.vscode/tasks.json file for the relevant command line arguments, e.g. from the project root directory:

```shell
python3 -m build . --wheel --sdist 
```

Note however that this requires setuptools >= 66.0.0 (version 59 is over 2 years old).

Any further queries, get back to me

**Comment by barower** [comment]
I've almost given up on trying to come up with solution, but then I had exact same issue with another library that stopped using `setup.py` recently. Building both with [poetry](https://github.com/python-poetry/poetry) finally worked, but I had to manually patch `pyproject.toml` files so that they have `[tool.poetry]` and `[tool.poetry.dependencies]` sections filled (FWIW this needs to be done only on older versions of `poetry`, for me that was the case because yocto's `kirkstone` branch is a little bit delayed release-wise)

By the way, if I understood [PEP517](https://peps.python.org/pep-0517/) correctly, your `build-backend` should be set to something else than `"setuptools.build_meta:__legacy__"`, since you don't have `setup.py`:

>If the `pyproject.toml` file is absent, or the `build-backend` key is missing, the source tree is not using this specification, and tools should revert to the legacy behaviour of running `setup.py` (either directly, or by implicitly invoking the `setuptools.build_meta:__legacy__` backend).

**Reply by semuadmin** [reply]
@barower Forgive me but I'm really not clear what the issue is here? The pyproject.toml file and folder structure follow the [PyPA recommendations](https://packaging.python.org/en/latest/tutorials/packaging-projects/) and I've had no reports of any issues building the package on any standard Python >=3.8 stack on any Linux, Windows or MacOS distribution _provided_ setuptools >=66.0.0. I've never had to resort to using poetry or any other Python dependency management or packaging framework.

What, specifically, are you looking for a solution to? What are you not able to achieve using the build and install commands I have provided?

Are you saying the command `python3 -m build . --wheel --sdist` does _not_ produce the requisite dist artefacts on yocto? If so, this sounds to me like an issue with yocto rather than this repo, but I'm happy to stand corrected.
